src/memline.c

```diff
@@ -2675,6 +2675,36 @@ ml_get_cursor(void)
 							curwin->w_cursor.col);
 }
 
+// return length (excluding the NUL) of the given line
+    colnr_T
+ml_get_len(linenr_T lnum)
+{
+    if (*ml_get_buf(curbuf, lnum, FALSE) == NUL)
+        return 0;
+
+    return curbuf->b_ml.ml_line_len - 1;
+}
+
+// return length (excluding the NUL) of the cursor line
+    colnr_T
+ml_get_curline_len(void)
+{
+    if (*ml_get_buf(curbuf, curwin->w_cursor.lnum, FALSE) == NUL)
+        return 0;
```

Comment:
I wonder if this can be tested

Reply:
I have not written a test before but I can give it a go. Any tips?